### PR TITLE
Artifact enable

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

File: engine/worker/cmd_server.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/ovh/cds/engine/log"
+)
+
+// Name of environment variable set to local worker HTTP server port
+// Used only to export build variables for now
+const WorkerServerPort = "CDS_EXPORT_PORT"
```
Comment (sguiheux) [inline engine/worker/cmd_server.go]: doc must begin with // WorkerServerPort

File: engine/worker/cmd_upload.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+func uploadHandler(w http.ResponseWriter, r *http.Request) {
+	// Get body
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var a sdk.Artifact
+	if err = json.Unmarshal(data, &a); err != nil {
```
Comment (sguiheux) [inline engine/worker/cmd_upload.go]: err :=

File: engine/worker/main.go [hunk 3]
```diff
@@ -156,7 +151,7 @@ func queuePolling() {
 	for {
 		if WorkerID == "" {
 			log.Notice("[WORKER] Disconnected from CDS engine, trying to register...\n")
-			err := register(api, name, UserKey)
+			err := register(api, name, key)
```
Comment (fsamin) [inline engine/worker/main.go]: one line !!!

File: engine/worker/cmd_upload.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if e := runArtifactUpload(a.Name, a.Tag, ab); e.Status != sdk.StatusSuccess {
```
Comment (sguiheux) [inline engine/worker/cmd_upload.go]: err :=

File: engine/worker/main.go (hunk 3 above)
Comment (sguiheux) [inline engine/worker/main.go]: one line

File: engine/worker/main.go [hunk 5]
```diff
@@ -297,7 +292,7 @@ func heartbeat() {
 		time.Sleep(10 * time.Second)
 		if WorkerID == "" {
 			log.Notice("[WORKER] Disconnected from CDS engine, trying to register...\n")
-			err := register(api, name, UserKey)
+			err := register(api, name, key)
```
Comment (sguiheux) [inline engine/worker/main.go]: one line

File: engine/worker/cmd_upload.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+		if errMarshal != nil {
+			sdk.Exit("internal error (%s)\n", errMarshal)
+		}
+
+		req, errRequest := http.NewRequest("POST", fmt.Sprintf("http://127.0.0.1:%d/upload", port), bytes.NewReader(data))
+		if errRequest != nil {
+			sdk.Exit("cannot post worker upload (Request): %s\n", errRequest)
+		}
+
+		resp, errDo := http.DefaultClient.Do(req)
```
Comment (fsamin) [inline engine/worker/cmd_upload.go]: set timeout on defaultclient

**Pull request merged**